### PR TITLE
fix: correctly handle nullable types of collections

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[X] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#784

### CODE REVIEW

**Comment by incendial** [comment]
@fzyzcjy hi, would you like to review a small fix to this rule?

File: lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_collection_methods_with_unrelated_types/avoid_collection_methods_with_unrelated_types_rule.dart [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import 'package:analyzer/dart/ast/ast.dart';
 import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/nullability_suffix.dart';
```
Comment (fzyzcjy) [inline lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_collection_methods_with_unrelated_types/avoid_collection_methods_with_unrelated_types_rule.dart:6]: Is it imported but not used?

**Review by fzyzcjy — COMMENTED** [review]
LGTM except a tiny bit

File: lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_collection_methods_with_unrelated_types/avoid_collection_methods_with_unrelated_types_rule.dart (hunk 1 above)
Comment (incendial) [reply lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_collection_methods_with_unrelated_types/avoid_collection_methods_with_unrelated_types_rule.dart:6]: That's how `part` and `part of` works - imports are listed only in "main" file. This one is used for `NullabilitySuffix.question`

Comment (fzyzcjy) [reply lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_collection_methods_with_unrelated_types/avoid_collection_methods_with_unrelated_types_rule.dart:6]: Oh I see. Only look at github review panel so not realized that

**Comment by incendial** [comment]
Having troubles with git, continuing here https://github.com/dart-code-checker/dart-code-metrics/pull/786

**Pull request closed without merging**